src/Universum.hs

```diff
@@ -1,14 +1,61 @@
 {-# LANGUAGE Trustworthy #-}
 
--- | Main module that reexports all functionality allowed to use
--- without importing any other modules. Just add next lines to your
--- module to replace default ugly 'Prelude' with better one.
---
--- @
---     {-\# LANGUAGE NoImplicitPrelude \#-}
---
---     import Universum
--- @
+{- | Main module that reexports all functionality allowed to use
+without importing any other modules. Just add next lines to your
+module to replace default 'Prelude' with better one.
+
+@
+\{\-\# LANGUAGE NoImplicitPrelude \#\-\}
+
+__import__ "Universum"
+@
+
+This documentation section contains description of internal module structure to
+help navigate between modules, search for interested functionality and
+understand where you need to put your new changes.
+
+Functions and types are distributed across multiple modules and grouped by
+meaning or __theme__. Name of the module should give you hints regarding what
+this module contains. Some /themes/ contains a lot of reexported stuff and a lot
```

Comment:
"Some /themes/ contain a great amount of both reexported functions and functions of our own. To make it easier to understand these huge chunks of functions, ... "